src/modules/orders/index.ts

```diff
@@ -67,13 +143,479 @@ class OrdersService {
+    this.inputCells.unshift({ previousOutput: biggestCell.outPoint!, since: '0x0' })
+    this.witnesses.unshift({
+      lock: '',
+      inputType: '',
+      outputType: '',
+    })
+
+    const dealMakerCell: CKBComponents.CellOutput = {
+      capacity: '0x' + (this.dealMakerCapacityAmount + BigInt(biggestCell.capacity)).toString(16),
+      lock: dealMakerLock,
+      type: this.outputsCells[0].type,
+    }
+    this.outputsCells.unshift(dealMakerCell)
+    this.outputsData.unshift(`0x${bigIntToUint128Le(this.dealMakerSudtAmount)}`)
+  }
+
+  private calculateMinerFee(rawTx: CKBComponents.RawTransactionToSign): bigint {
+    return BigInt(getTransactionSize(rawTx)) * this.feeRatio
+  }
+
+  private subMinerFeeAndUpdateOutputs(
+    rawTx: CKBComponents.RawTransactionToSign,
+    minerFee: bigint,
+  ): CKBComponents.RawTransactionToSign {
+    const dealMakerOutput: CKBComponents.Cell = rawTx.outputs[0]
+    const dealMakerCapacity = '0x' + (BigInt(rawTx.outputs[0].capacity) - minerFee).toString(16)
+    const newDealMakerOutput: CKBComponents.Cell = { ...dealMakerOutput, ...{ capacity: dealMakerCapacity } }
+    this.outputsCells.shift()
+    this.outputsCells.unshift(newDealMakerOutput)
+    return { ...rawTx, ...{ outputs: this.outputsCells } }
+  }
+
+  private generateRawTx() {
+    const rawTransaction: CKBComponents.RawTransactionToSign = {
+      version: '0x0',
+      headerDeps: [],
+      cellDeps: [
+        {
+          outPoint: {
+            txHash: SUDT_TX_HASH,
+            index: '0x0',
+          },
+          depType: 'code',
+        },
+        {
+          outPoint: { txHash: ORDERBOOK_TX_HASH, index: '0x0' },
+          depType: 'code',
+        },
+        {
+          outPoint: { txHash: SECP256K1_TX_HASH, index: '0x0' },
+          depType: 'depGroup',
+        },
+      ],
+      inputs: this.inputCells,
+      witnesses: this.witnesses,
+      outputs: this.outputsCells,
+      outputsData: this.outputsData,
+    }
+
+    return rawTransaction
+  }
+
+  private generateDealStruct(minerFee: bigint, arg: string) {
+    const orderIds: string = this.inputCells
+      .map((input: CKBComponents.CellInput) => `${input.previousOutput!.txHash}-${input.previousOutput!.index}`)
+      .join()
+    const fee: string = `${this.dealMakerCapacityAmount - minerFee}-${this.dealMakerSudtAmount}`
+    return {
+      txHash: '',
+      tokenId: arg,
+      orderIds: orderIds,
+      fee: fee,
+      status: DealStatus.Pending,
+    }
+  }
+
+  private async sendTransactionAndSaveDeal(
+    rawTransaction: CKBComponents.RawTransactionToSign,
+    lock: CKBComponents.Script,
+    privateKey: string,
+    deal: { txHash: string; status: number; orderIds: string; fee: string; tokenId: string },
+  ) {
+    try {
+      const response = await signAndSendTransaction(rawTransaction, privateKey, lock)
+      deal.txHash = response
+
+      console.info('==============================')
+      console.info(response)
```

Comment:
Use `logger` to print info with prefix
Ref: https://github.com/glias/deal-maker-cli/blob/develop/src/modules/orders/index.ts#L15-L17